arch/Kconfig

```diff
@@ -413,6 +413,12 @@ config MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
 	  This option is enabled when the MPU requires a power of two alignment
 	  and size for MPU regions.
 
+config MPU_REQUIRES_NON_OVERLAPPING_REGIONS
+	bool
```

Comment:
@ruuddw likely useful for ARC as well, the V3 ARC MPU also doesn't allow overlaps